test/api/exportAssertion.spec.js

```diff
@@ -0,0 +1,229 @@
+                expect(subject, 'to equal', 'foo');
+            });
+        });
+
+        it('should succeed', function () {
+            parentExpect('', 'when prepended with foo', 'to foo');
+        });
+
+        it('should fail with a diff', function () {
+            expect(function () {
+                parentExpect('bar', 'when prepended with foo', 'to foo');
+            }, 'to throw',
+                "expected 'bar' when prepended with foo to foo\n" +
+                "\n" +
+                "-foobar\n" +
+                "+foo"
+            );
+        });
+    });
+
+    describe('when shifting from an exported middle-of-the-rocket assertion in an inherited expect to an assertion exported from another inherit', function () {
+        beforeEach(function () {
+            inheritedExpect.exportAssertion('<string> when prepended with foo <assertion?>', function (expect, subject) {
+                return expect.shift('foo' + subject);
+            });
+
+            parentExpect.inherit().exportAssertion('<string> to foo', function (expect, subject) {
+                expect(subject, 'to equal', 'foo');
+            });
+        });
+
+        it('should succeed', function () {
+            parentExpect('', 'when prepended with foo', 'to foo');
+        });
+
+        it('should fail with a diff', function () {
+            expect(function () {
+                parentExpect('bar', 'when prepended with foo', 'to foo');
+            }, 'to throw',
+                "expected 'bar' when prepended with foo to foo\n" +
+                "\n" +
+                "-foobar\n" +
+                "+foo"
+            );
+        });
+    });
+
+
+    describe('when delegating to an exported assertion from the parent', function () {
+        beforeEach(function () {
+            inheritedExpect.exportAssertion('<string> when prepended with foo <assertion?>', function (expect, subject) {
+                return expect.shift('foo' + subject);
+            });
+
+            inheritedExpect.exportAssertion('<string> when trimmed <assertion?>', function (expect, subject) {
+                return expect.shift(subject.trim());
+            });
+
+            parentExpect.inherit().exportAssertion('<string> to foo', function (expect, subject) {
+                expect(subject, 'to equal', 'foo');
+            });
+        });
+
+        it('should succeed', function () {
+            parentExpect('', 'when prepended with foo when trimmed', 'to foo');
+        });
+
+        it('should fail with a diff', function () {
+            expect(function () {
+                parentExpect('bar', 'when prepended with foo when prepended with foo', 'to foo');
+            }, 'to throw',
+                "expected 'bar' when prepended with foo when prepended with foo to foo\n" +
+                "\n" +
+                "-foofoobar\n" +
+                "+foo"
+            );
+        });
+    });
+});
```

Comment:
In case I didn't actually overlooked something, it looks like we don't test that non-exported assertions, styles and types are not actually leaking. That is huge important in my opinion.